Pivotgrid revision

silviyaboteva opened the pull request:
Closes: https://github.com/telerik/kendo-themes/issues/2498

There are two things that are discussed with the design team and they will be updated in the designs later:
- Bootstrap theme: The checkboxes in the Treeview component have a white background in an unchecked state.
- The darker border for the total column will be removed
- The darker border separating the different fields in the row/column headers will also be removed.

Screenshot for the 
<img width="1607" alt="Screenshot 2021-07-19 at 14 16 04" src="https://user-images.githubusercontent.com/17604855/126307981-0a723b03-2a44-40ae-a951-03fefb6702f1.png">

cc/ @ag-petrov @kspeyanski @nstoychev

Juveniel commented on packages/default/scss/pivotgrid/_layout.scss:
```diff
@@ -1,4 +1,453 @@
+    }
+
+    .k-pivotgrid-configurator-actions {
+        margin: (-$pivotgrid-configurator-actions-padding-y) (-$pivotgrid-configurator-actions-padding-x);
```
we can remove this negative margin and instead utilize the `k-actions k-hstack k-justify-content-end` classes for the container. See https://github.com/telerik/kendo-themes/commit/9ff8244e9d17b6b473990b00d680548d67248f82 for reference.  

This will also make the spacing in the container consistent, as the currently, the actions wrapper have padding and receive additional padding from the parent container.

silviyaboteva replied:
Thank you for noticing this. I changed the action buttons in the column menus, but forget to do the same for configurator itself.

Juveniel commented on packages/material/scss/pivotgrid/_theme.scss:
```diff
@@ -1,4 +1,19 @@
-@include exports("pivotgrid/theme") {
+@import "~@progress/kendo-theme-default/scss/pivotgrid/_theme.scss";
+
+@include exports("pivotgrid/theme/material") {
+    // Hover and Focus state have the same bg color
```
Add spacing rows here for consistency

Juveniel commented:
@silviyaboteva I have also noticed that 

1) After running PostCSS Sorting in the _layout file, there are some changes in the properties order (not all of which seem legit). You could check on your end and apply the sorting where possible,

2) I have also noticed that in the configurator container there is no padding, thus when focusing inputs/buttons the top outline will not be visible. We should probably discuss this with the design team and adjust the spacing:

![image](https://user-images.githubusercontent.com/16972160/126446735-194a5f41-4580-4236-a7f7-d2807d117f79.png)

silviyaboteva commented on packages/default/scss/pivotgrid/_layout.scss:
```diff
@@ -1,4 +1,453 @@
+
+
+    // Modes
+    .k-pivotgrid-configurator-push {
+        .k-pivotgrid-configurator-vertical {
+            min-width: $pivotgrid-configurator-vertical-width;
+        }
+    }
```
This, in fact, is a leftover. I'll remove it.

silviyaboteva commented:
Hey @Juveniel 
Thank you for the extensive review. I addressed the comments in the code. About the general comment:
1. I updated the properties order in all files as well.
2. I synced with @ivelina-penkova and we agreed to add 2px (for Default theme) and 0.25rem (for Bootstrap theme) to the `$pivotgrid-configurator-content-padding-y` variable in order to prevent cutting the focus shadow of the inputs/buttons. Other themes don't apply top shadow.

joneff commented on packages/bootstrap/scss/pivotgrid/_layout.scss:
```diff
@@ -1 +1,9 @@
+@include exports ("pivotgrid/layout/bootstrap/legacy") {
+
+    .k-pivot-toolbar {
+        padding: $toolbar-padding-y;
```
if we want to override a variable, we create a local one. In this case, two local ones: `$pivotgrid-toolbar-padding-x` and `$pivotgrid-toolbar-padding-y` or we don't customize the padding at all

silviyaboteva replied:
Vanka, this line of code is inside the legacy styles of the Pivotgrid. I didn't touch the styles there, only place them under `/legacy` module. In the current rendering, there is no toolbar at all.

silviyaboteva replied:
I noticed that in `https://github.com/telerik/kendo-themes/pull/2606`, you've removed these pivot toolbar styles. It might be my mistake when I rebased. I'll erase them.

joneff commented on packages/bootstrap/scss/pivotgrid/_theme.scss:
```diff
@@ -1,4 +1,6 @@
-@include exports("pivotgrid/theme") {
+@import "~@progress/kendo-theme-default/scss/pivotgrid/_theme.scss";
+
+@include exports("pivotgrid/theme/legacy") {
```
why are we keeping the legacy styling?

silviyaboteva replied:
Due to the fact that jQuery still uses it and if we merge the PR before they change their rendering with the new one, it will broke their widget. Once they apply the new rendering, we can safely remove the legacy modules.

silviyaboteva replied:
Also, they will have both renderings for a while. So, we need the styles for now.

joneff commented on packages/default/scss/pivotgrid/_theme.scss:
```diff
@@ -1,5 +1,170 @@
 @include exports("pivotgrid/theme") {
 
+    .k-pivotgrid,
+    .k-pivotgrid-values {
```
no need for the second selector.

joneff commented on packages/default/scss/pivotgrid/_theme.scss:
```diff
@@ -1,5 +1,170 @@
+        );
+    }
+
+    .k-pivotgrid-empty-cell,
+    .k-pivotgrid-column-headers,
+    .k-pivotgrid-row-headers {
+        @include fill (
+            $color: $pivotgrid-headers-text,
+            $bg: $pivotgrid-headers-bg
+        );
+    }
+
+    .k-pivotgrid-column-headers,
+    .k-pivotgrid-row-headers {
+        @include fill (
+            $border: $pivotgrid-headers-border
+        );
+    }
```
consider combining the two selectors. you are setting empty cell border in the next line, so it will override the previous setting

joneff commented on packages/default/scss/treeview/_layout.scss:
```diff
@@ -98,6 +98,10 @@
             margin-left: -$treeview-indent;
             cursor: pointer;
+
+            + .k-checkbox-wrapper {
```
consider extracting this as a separate fix

joneff reviewed (changes requested):
Overall I like the rendering. I am still thinking on the styles
